nni/contrib/compression/distillation/basic_distiller.py

```diff
@@ -0,0 +1,342 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+
+from collections import defaultdict
+from copy import deepcopy
+import logging
+from typing import Any, Callable, Dict, List, overload
+
+import torch
+import torch.nn.functional as F
+from torch.optim import Adam
+
+from ..base.compressor import Compressor, Distiller, _DISTILLATION_TARGET_SPACES
+from ..base.wrapper import ModuleWrapper, register_wrappers
+from ..utils.evaluator import Evaluator
+
+_logger = logging.getLogger(__name__)
+
+
+class TeacherModelBasedDistiller(Distiller):
+    """
+    The base class that the distiller need a teacher model.
+
+    Parameters
+    ----------
+    model
+        The student model to be distilled.
+    config_list
+        A list of dict, each dict configure which module need to be distilled, and how to distil.
+        Please refer :doc:`Compression Config Specification </compression/compression_config_list>` for more information.
+    evaluator
+        Please refer TODO.
```

Comment:
still TODO here?

Reply:
yes, will update for all compressor after evaluator doc done...